pkg/db-repo/repository_test.go

```diff
@@ -675,6 +676,151 @@ func TestRepository_DeleteHasManyNoRelation(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+type gormSetTypeValid struct {
+	T    []uint8
+	TSet []string
+}
+
+type gormSetTypeInvalid1 struct {
+	T    []uint8
+	TSet string
+}
+
+type gormSetTypeInvalid2 struct {
+	T    []string
+	TSet []string
+}
+
+type gormSetTypeInvalid3 struct {
+	TSet []string
+}
+
+func TestGormParseMysqlSetType(t *testing.T) {
+	assert.NotPanics(t, func() {
+		scope := gorm.Scope{}
+		db_repo.GormParseMysqlSetType(&scope)
+	}, "should not panic on nil")
+	assert.NotPanics(t, func() {
+		scope := gorm.Scope{
+			Value: &gormSetTypeValid{
+				T:    []uint8{97, 44, 98, 44, 99},
+				TSet: make([]string, 0),
+			},
+		}
+		db_repo.GormParseMysqlSetType(&scope)
+		assert.Equal(t, []string{"a", "b", "c"}, scope.Value.(*gormSetTypeValid).TSet)
+	}, "should not panic on a,b,c in test struct")
+	assert.NotPanics(t, func() {
+		scope := gorm.Scope{
+			Value: &gormSetTypeValid{
+				T:    nil,
+				TSet: make([]string, 0),
+			},
+		}
+		db_repo.GormParseMysqlSetType(&scope)
+		assert.Equal(t, []string{}, scope.Value.(*gormSetTypeValid).TSet)
+	}, "should not panic on a,b,c in test struct")
+	assert.NotPanics(t, func() {
+		scope := gorm.Scope{
+			Value: &gormSetTypeValid{
+				T:    nil,
+				TSet: nil,
+			},
+		}
+		db_repo.GormParseMysqlSetType(&scope)
+		assert.IsType(t, []string{}, scope.Value.(*gormSetTypeValid).TSet)
+	}, "should not panic on a,b,c in test struct")
```

Comment:
Same again. Also, isn't `assert.IsType(t, []string{}, scope.Value.(*gormSetTypeValid).TSet)` always true because `TSet` is declared to have that type?